src/main/java/se/liu/ida/hefquin/engine/data/impl/VocabularyMappingImpl.java

```diff
@@ -61,20 +61,19 @@ public SPARQLGraphPattern translateTriplePattern(final TriplePattern tp) {
 		List<SPARQLGraphPattern> predicateTranslation = Collections.<SPARQLGraphPattern>emptyList();
 		for(final SPARQLGraphPattern j : objectTranslation) {
 			
-			if(j instanceof SPARQLUnionPattern) { 
-				List<SPARQLGraphPattern> unionList = Collections.<SPARQLGraphPattern>emptyList();
+			if(j instanceof SPARQLUnionPattern) {
+				SPARQLUnionPatternImpl union = new SPARQLUnionPatternImpl();
```

Comment:
I found that for SPARQLUnionPattern specifically, I hade to use the Impl version to be able to call the addSupPattern function. Should I access this function another way?

Reply:
Just some clarification: While Sebastian is right in general, in this specific case it is indeed correct to use `SPARQLUnionPatternImpl` rather than `SPARQLUnionPattern`, because `addSubPattern` is not declared in the interface but only in the implementing class. See my earlier comment: https://github.com/LiUSemWeb/HeFQUIN/pull/162#issuecomment-1048895270

Now you may ask why I did not add the `addSubPattern` function also to the interface. The reason is that I would generally like these interfaces to represent immutable objects.